File: QuSoC/QuSoC/apps/Recursion/firmware/Recursion.cpp
#include "Recursion.h"
#include <stdint.h>
#include <stdbool.h>
#include "soc.h"
namespace Recursion
{
	void Firmware::RecursiveCounter(uint32_t counter)
	{
		Recursion_SOC_Counter += counter;
		if ((counter < 10))
		{
			RecursiveCounter((counter + 1));
		}
	}
	void Firmware::EntryPoint()
	{
		RecursiveCounter(0);
	}
}

File: QuSoC/QuSoC/apps/Recursion/source/Recursion.cs
namespace Recursion
{
    public class SOC
    {
        public static SOC Instance { get; set; } = new SOC();

        public virtual uint Counter { get; set; }
    }

    public static class Firmware
    {
        static void RecursiveCounter(uint counter)
        {
            SOC.Instance.Counter += counter;

            if (counter < 10)
                RecursiveCounter(counter + 1);
        }

        public static void EntryPoint()
        {
            RecursiveCounter(0);
        }
    }
}
